### DISCUSSION TITLE
Integrating Kotlin/Native project with SonarQube

### DISCUSSION THREAD

**Discussion by NorseDreki** [z4kn4fein/kotlin-semver#45] [General]
Hello there,

Just wanted to ask an off topic question -- were you able to integrate this project with SonarQube analysis? I see it uses such an integration. For my own Kotlin/Native project, SonarQube shows 0 source code lines.

So I'm looking for a success story or for some clues in this area.

Pardon if this is irrelevant.

**Comment by z4kn4fein** [comment]
Hi @NorseDreki,

Yes, this project is integrated with SonarCloud using `detekt` as a code analyzer. The CI workflow runs the [`detekt` analysis](https://github.com/z4kn4fein/kotlin-semver/blob/main/.github/workflows/semver-ci.yml#L50) and runs sonar with the report path in the [sonar properties](https://github.com/z4kn4fein/kotlin-semver/blob/main/build.gradle.kts#L195-L207). 
When this project was created, Sonar did not support Kotlin, this is why a `detekt` report is uploaded. Now, it seems there's [built-in support for Kotlin Multiplatform](https://community.sonarsource.com/t/kotlin-multiplatform-support-is-here/92577) projects, but I haven't tried it yet.

I hope this helps, let me know if you have further questions!
